pkg/compact/compact.go

```diff
@@ -275,6 +280,15 @@ func (c *Syncer) downloadMeta(ctx context.Context, id ulid.ULID) (*metadata.Meta
 		return nil, errors.Wrapf(err, "downloading meta.json for %s", id)
 	}
 
+	// Check for block labels by relabeling.
```

Comment:
Nice. However I would move it out of `downloadMeta` as we do more than method suggests which might be confusing. Let's do it after meta is downloaded? Also it's nice to do it after downloadMeta method as we check if it's not too fresh first.